lib/services/elasticsearch.js

```diff
@@ -153,519 +166,479 @@ class ElasticSearch extends Service {
-          && !request.input.args.includeTrash
-        ) {
-          return this.reject('document_not_found', esRequest.id);
-        }
+      .then(({ body }) => {
+        const apiResponse = {
+          _id: body._id,
+          _version: body._version,
+          _source: body._source
+        };
 
-        return prepareMetadata(result);
+        return apiResponse;
       })
-      .catch(error => Bluebird.reject(this.esWrapper.formatESError(error)));
+      .catch(error => this.esWrapper.reject(error));
   }
 
   /**
    * Return the list of documents matching the ids given in the body param
    * NB: Due to internal Kuzzle mechanism, can only be called on a single index/collection,
    * using the body { ids: [.. } syntax.
-   * @param {Request} request
-   * @returns {Promise}
+   *
+   * @param {string} index - Index name
+   * @param {string} collection - Collection name
+   * @param {Array<string>} ids - Document IDs
+   *
+   * @returns {Promise} { hits, total }
    */
-  mget(request) {
-    const esRequest = initESRequest(request);
-
-    esRequest.body = request.input.body;
-    delete esRequest.body.includeTrash;
+  mget (index, collection, ids) {
+    const esRequest = {
+      index: this._getESIndex(index, collection),
+      body: ids
+    };
 
     return this.client.mget(esRequest)
-      .then(result => {
-        // harmonize response format based upon the search one
-        if (result.docs) {
-          result.hits = request.input.args.includeTrash
-            ? result.docs
-            : result.docs.filter(
-              doc => doc._kuzzle_info === undefined ||
-              doc._kuzzle_info.active
-            );
-          delete result.docs;
-
-          result.hits = result.hits.map(obj => prepareMetadata(obj));
-        }
+      .then(({ body }) => {
+        const apiResponse = {
+          hits: body.docs,
+          total: body.total
+        };
 
-        return result;
+        return apiResponse;
       })
-      .catch(error => Bluebird.reject(this.esWrapper.formatESError(error)));
+      .catch(error => this.esWrapper.reject(error));
   }
 
   /**
    * Count how many documents match the filter given in body
-   * @param {Request} request
-   * @returns {Promise} resolve the number of document
+   *
+   * @param {string} index - Index name
+   * @param {string} collection - Collection name
+   * @param {object} query - Query to match
+   *
+   * @returns {Promise} { count }
    */
-  count(request) {
-    const esRequest = initESRequest(request);
-
-    esRequest.body = request.input.body;
+  count (index, collection, query = {}) {
+    const esRequest = {
+      index: this._getESIndex(index, collection),
+      body: query
+    };
 
-    if (!request.input.args.includeTrash) {
-      addActiveFilter(esRequest);
-    }
     return this.client.count(esRequest)
-      .catch(error => Bluebird.reject(this.esWrapper.formatESError(error)));
+      .then(({ body }) => {
+        const apiResponse = {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Immediately return this expression instead of assigning it to the temporary variable "apiResponse". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1488)